src/lib/components/Icon/Icon.svelte

```diff
@@ -1,31 +1,27 @@
+<svelte:options immutable />
+
 <script lang="ts">
   import classNames from "$lib/util/classNames";
-  import icons from "./icons";
-  type IconName = Extract<keyof typeof icons, string>;
 
-  export let name: IconName;
   // Size is taken from the USWDS implemention of icons in their icon component package (see
   // https://designsystem.digital.gov/components/icon/#using-the-icon-component-2 for more details).
   // Reason for undefined `size` value to start: "By default, icons will scale with font size.
   // If you want to specify an icon size, use one of the component’s size variants."
   export let size: 3 | 4 | 5 | 6 | 7 | 8 | 9 | undefined = undefined;
-  export let className = "";
 
-  // As a default value, make the alt just a more titleized version of the file name,
-  // e.g., "arrow_upward" becomes "Arrow Upward"
-  export let alt = name
-    .split("_")
-    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
-    .join(" ");
+  let className = "";
+  export { className as class };
+
+  export let alt = "";
+  export let src: string;
+  export let plain = false;
 
-  if (!icons[name]) {
-    throw new Error(`${name} is not a valid icon!`);
-  }
+  if (plain && size) console.warn('"size" prop has no effect when "plain" is set');
```

Comment:
Might be mistaken, but isn't `plain = true` the same as `size = undefined` currently? If so I feel like it will be easier to omit `size` from props rather than remember to add `plain`.

Reply:
`plain` is not the same as `size={undefined}`. `plain` is used to omit _all_ the default classes including `usa-icon`, whereas `size={undefined}` only omits `usa-icon--size-SIZE`.

Reply:
Ah okay, I missed that `plain` would omit `usa-icon` when reading. That makes more sense then, since there are cases where we want the SVG icon but don't want USWDS to apply the icon styling.